contracts/src/v0.8/ccip/pools/SiloedLockReleaseTokenPool.sol

```diff
@@ -0,0 +1,286 @@
+  /// the amount will be the amount of liquidity shared among all unsiloed chains.
+  function getSiloedTokensByChain(
+    uint64 remoteChainSelector
+  ) external view returns (uint256 lockedTokens) {
+    if (s_siloedChainConfigs[remoteChainSelector].isSiloed) {
+      return s_siloedChainConfigs[remoteChainSelector].siloedBalance;
+    }
+
+    return s_unsiloedTokenBalance;
+  }
+
+  /// @notice Returns the amount of tokens in the token pool that are shared among all unsiloed chains.
+  /// @return unsiloedTokens amount of tokens available to all unsiloed chains.
+  function getliquidityForUnsiloedChains() public view returns (uint256) {
+    return s_unsiloedTokenBalance;
+  }
+
+  /// @notice Updates designations for chains on whether to mark funds as Siloed or not
+  /// @param removes A list of chain selectors to disable Siloing. Their funds will be moved into the unsiloed pool.
+  /// @param adds A list of chain selectors to enable Siloing. Adding a chain to siloing will not set the rebalancer.
+  /// The rebalancer will need to be set separately.
+  function updateSiloDesignationForChainSelectors(
+    uint64[] calldata removes,
+    ChainSiloConfigUpdate[] calldata adds
+  ) external onlyOwner {
+    for (uint256 i = 0; i < removes.length; ++i) {
+      // When a chain is removed from siloing, the funds are moved to the accounting pool shared by all unsiloed chain.
+      uint256 amountUnsiloed = s_siloedChainConfigs[removes[i]].siloedBalance;
+
+      s_unsiloedTokenBalance += amountUnsiloed;
+
+      /// @notice Removing the entire configuration will also delete the rebalancer. If the chain ever becomes siloed
+      /// again, the rebalancer will need to be set again.
+      delete s_siloedChainConfigs[removes[i]];
+
+      // Emit a removal event which includes the amount of funds moved to the general silo.
+      emit ChainUnsiloed(removes[i], amountUnsiloed);
+    }
+
+    for (uint256 i = 0; i < adds.length; ++i) {
+      /// @notice Adding a chain to siloing will not set the rebalancer
+      s_siloedChainConfigs[adds[i].remoteChainSelector].isSiloed = true;
+      s_siloedChainConfigs[adds[i].remoteChainSelector].rebalancer = adds[i].rebalancer;
+
+      // Emit a siloing event to indicate that the chain is now siloed, but that the amount of liquidity
+      // is zero until liquidity is provided.
+      emit ChainSiloed(adds[i].remoteChainSelector);
+    }
+  }
+
+  /// @notice Gets the rebalancer able to provide liquidity for a remote chain selector
+  /// @param remoteChainSelector The CCIP specific selector for the remote chain being interacted with.
+  /// @return The current liquidity manager, contract owner if the chain's funds are not siloed.
+  function getRebalancerByChain(
+    uint64 remoteChainSelector
+  ) public view returns (address) {
+    ChainSiloConfig memory remoteConfig = s_siloedChainConfigs[remoteChainSelector];
+    if (remoteConfig.isSiloed) return remoteConfig.rebalancer;
```

Comment:
I know this is perfectly legal Solidity but let's add `{ }` and remove the else. No else needed if the `if` returns